llama-index-core/llama_index/core/program/function_program.py

```diff
@@ -51,6 +67,101 @@ def model_fn(**kwargs: Any) -> BaseModel:
     )
 
 
+# def create_partial_model(model: Type[BaseModel], processed_models=None) -> Type[BaseModel]:
```

Comment:
do we need this or nah?

Reply:
nope this was my first attempt with ChatGPT to create a function, reverted to the more simpler one below